scripts/cli/src/util/sqlite3.ts

```diff
@@ -0,0 +1,39 @@
+import sql from "sqlite3";
+import type { KeyringPair } from "@polkadot/keyring/types";
+
+const sql3 = sql.verbose();
+// Setting up a database for storing data.
+const db = new sql3.Database("database.db");
+
+export function createTable() {
+  db.run(
+    "CREATE TABLE IF NOT EXISTS next_nonce ( account TEXT PRIMARY KEY, nonce INT DEFAULT 0 )"
+  );
+}
+
+export async function getNonce(signer: KeyringPair) {
+  return new Promise<number>((resolve, reject) => {
+    db.run(
+      "INSERT INTO next_nonce(account) VALUES($account) ON CONFLICT(account) DO UPDATE SET nonce=nonce+1",
+      { $account: signer.address },
+      (err) => {
+        if (err) {
+          console.log(err);
+        } else {
+          db.get(
+            "SELECT account, nonce FROM next_nonce WHERE account = $account",
```

Comment:
It looks like new versions of SQLite have `RETURNING` now:

https://dev.to/mliakos/quick-post-about-the-sqlite-upsert-and-the-new-returning-clause-5fhl

You should be able to add `RETURNING *` to the previous query instead of needing to do a second call.

Reply:
I had looked into it but wasn't able to get it working at the moment there weren't any examples of using it with the db.run so extracting the value was a problem for me.